### PR TITLE
Make 'splitAtSign' more consistent

### LINKED ISSUE #1808: Make splitAtSign consistent
The behaviour of ``splitAtSign`` is not consistent between the endpoint /validate/check and /auth.
We should adapt the endpoint /auth this way, that it behaves in the same was as the endpoint /validate/check.

### PR DESCRIPTION
**Pull request by plettich**
The `splitAtSign` configuration did not affect the `/auth` endpoint thus
leading to inconsistencies.
This PR tries to mimic the user-resolving behaviour of the
`/validate/check` endpoint considering the `splitAtSign` configuration.

Fixes #1808

### CODE REVIEW

File: tests/test_api_validate.py [hunk 1]
```diff
@@ -747,22 +747,6 @@ def test_05_check_serial_with_no_user(self):
             self.assertEqual(result.get("status"), True)
             self.assertEqual(result.get("value"), True)
 
-    def test_05_check_serial_with_no_user(self):
```
Comment (cornelinux) [inline tests/test_api_validate.py]: Why did you remove this test?

File: tests/test_api_auth.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
+# -*- coding: utf-8 -*-
+""" Test for the '/auth' API-endpoint """
+
+from .base import MyApiTestCase
+import mock
+import six
+from privacyidea.lib.config import set_privacyidea_config
+
+
+class AuthApiTestCase(MyApiTestCase):
```
Comment (cornelinux) [inline tests/test_api_auth.py]: Could you please add a test case for an internal admin?
We have this one "testadmin"/"testpw", but in theory you could also define an internal admin like "admin@int", even if the "int" realm does not exist. And I think the code will and should work, since we do not raise an error (which is fine) the pseudo realm "int" would be split and the admin is verify with the complete ``username`` "admin@int".

File: tests/test_lib_user.py [hunk 3]
```diff
@@ -199,7 +202,27 @@ def test_09_get_user_from_param(self):
                  "realm": self.realm2}
         user = get_user_from_param(param)
         self.assertEqual("{0!s}".format(user), "<cornelius.resolver1@realm2>")
-        
+
+        # test with splitAtSign set to '0'
```
Comment (cornelinux) [inline tests/test_lib_user.py]: Why did you remove this one? Did it change or did you move it to the other tests?
I think then we can also delete it - not only comment it.

**Comment by cornelinux** [comment]
I added an issue #1908 for the documentation. So my generic comment is obsolete.
Only see my three comments inline.

File: tests/test_api_validate.py (hunk 1 above)
Comment (plettich) [reply tests/test_api_validate.py]: Ah, the github diff doesn't show but this function is a duplicate. The same function name with the same code is used just before this one :-)

File: tests/test_api_auth.py (hunk 2 above)
Comment (plettich) [reply tests/test_api_auth.py]: will do

File: tests/test_api_validate.py (hunk 1 above)
Comment (cornelinux) [reply tests/test_api_validate.py]: thanks.

File: tests/test_lib_user.py (hunk 3 above)
Comment (plettich) [reply tests/test_lib_user.py]: This is a remnant from when i thought we could fix this all in one place...
But actually, the `splitAtSign` configuration isn't tested here (it is used in `get_user_from_param()`, not in `splig_user()`) so i would add some tests for the library functions as well.

File: tests/test_api_auth.py (hunk 2 above)
Comment (plettich) [reply tests/test_api_auth.py]: Done in 3bcd19af

File: tests/test_lib_user.py (hunk 3 above)
Comment (plettich) [reply tests/test_lib_user.py]: removed the commented code and added tests for `splitAtSign` in 3bcd19af
